### PR TITLE
Enable rule

### PR DESCRIPTION
**Pull request by TomasVotruba**
Follow up to https://github.com/rectorphp/rector-src/pull/2618

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@samsonasik Now it's good to review :+1:  :slightly_smiling_face: 

I've merged the raw rule here https://github.com/rectorphp/rector-src/pull/2618, so the fixes and bugs are clearly separated.

**Comment by samsonasik** [comment]
Various things be skipped/merged:

1. indirect next used should be skipped:

```php
$result = execute();
$this->temporaryValue = $result;

otherCode();
otherCode(); 

function () use ($result) {

};

echo $result;
```

2. with `@var` usage, it should be skipped, eg:
----

```php
/** @var string $result */
$result = execute();
$this->temporaryValue = $result;
```

3. with comment, the comments should be merged, eg:
---

```php
// some comment
$result = execute();

// another comment
$this->temporaryValue = $result;
```

should be:

```php
// some comment
// another comment
$this->temporaryValue = execute();
```

**Comment by TomasVotruba** [comment]
I see. Could you handle these edge-cases?

**Comment by samsonasik** [comment]
I will try.

File: config/set/early-return.php [hunk 1]
```diff
@@ -31,6 +31,6 @@
         PreparedValueToEarlyReturnRector::class,
         ReturnBinaryOrToEarlyReturnRector::class,
         ReturnEarlyIfVariableRector::class,
-        // RemoveJustVariableAssignRector::class,
+        RemoveJustVariableAssignRector::class,
```
Comment (samsonasik) [inline config/set/early-return.php:34]: `RemoveJustVariableAssignRector` should be in dead-code instead of early return set

**Comment by samsonasik** [comment]
https://github.com/rectorphp/rector-src/pull/2621

File: config/set/early-return.php (hunk 1 above)
Comment (TomasVotruba) [reply config/set/early-return.php:34]: Yes, you're right :+1: I forgot the old category

Comment (samsonasik) [reply config/set/early-return.php:34]: moved at https://github.com/rectorphp/rector-src/pull/2621

**Pull request merged**